### PR TITLE
feat: implemented route for event component check

### LINKED ISSUE #6205: Server endpoints for Events & related attributes check
**Is your feature request related to a problem? Please describe.**
There should be an endpoint which tells if various components of an event are available. For instance, if sessions or tickets are available. Currently, just to check if sessions or tickets exist, a significant number of them are loaded from the backend which ruins the performance of frontend.

**Describe the solution you'd like**
Endpoints must be implemented to check for related properties

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6205 

#### Short description of what this resolves:
 Adds a route to check for availability of event components to omit costly server requests.

#### Changes proposed in this pull request:

- Added component checks for speakers, sessions & tickets

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
> Not good, implement as event relationship so that clients can just include it with event instead of hitting another endpoint

Screenshot 2019-07-17 at 12 41 30 PM

These already seem to be implemented. Do we require anything else?

**Comment by iamareebjamal** [comment]
I thought you just wanted boolean status

`GET event/21/?include=event-info`

```json
{
 ... # Other event properties
 "event-info": {
    "has-speakers": true,
    "has-tickets": false
  }
}
```

**Comment by mrsaicharan1** [comment]
> I thought you just wanted boolean status
> 
> `GET event/21/?include=event-info`
> 
> ```json
> {
>  ... # Other event properties
>  "event-info": {
>     "has-speakers": true,
>     "has-tickets": false
>   }
> }
> ```


Yes, that's what is required. How are relationships like that usually implemented?

**Comment by iamareebjamal** [comment]
See any relationship and the docs of flask-json-restapi

**Comment by mrsaicharan1** [comment]
> I thought you just wanted boolean status
> 
> `GET event/21/?include=event-info`
> 
> ```json
> {
>  ... # Other event properties
>  "event-info": {
>     "has-speakers": true,
>     "has-tickets": false
>   }
> }
> ```
> 
> If you just want all the tickets and sessions and speakers, just include them in the endpoint. However, be warned, it's going to take a long time to fetch if you load everything in one call


@iamareebjamal 
I found one of these ideas on the ideas page.. I thought the current implementation would solve it.
Screenshot 2019-07-17 at 5 27 23 PM

**Comment by iamareebjamal** [comment]
No, you are still returning all tickets and speakers, and this is not JSONAPI spec compliant

**Comment by iamareebjamal** [comment]
Please create relationshiips

File: app/models/event.py [hunk 1]
```diff
@@ -35,6 +35,18 @@ def get_new_event_identifier(length=8):
...
+class EventData(db.Model):
+    __tablename__ = 'event_data'
+    id = db.Column(db.Integer, primary_key=True)
+
+    has_speakers = db.Column(db.Boolean, default=False, nullable=False)
+    has_sessions = db.Column(db.Boolean, default=False, nullable=False)
+    has_tickets = db.Column(db.Boolean, default=True, nullable=False)
+
+    event_id = db.Column(db.Integer, db.ForeignKey('events.id', ondelete='CASCADE'))
+    event = db.relationship('Event', backref='event_info', foreign_keys=[event_id])
```
Comment (iamareebjamal) [inline app/models/event.py]: No need of this

File: app/models/event.py [hunk 2]
```diff
@@ -72,6 +84,7 @@ class Event(SoftDeletionModel):
     speaker = db.relationship('Speaker', backref="event")
     sponsor = db.relationship('Sponsor', backref="event")
     tickets = db.relationship('Ticket', backref="event_")
+    event_data = db.relationship('EventData', backref="event_")
```
Comment (iamareebjamal) [inline app/models/event.py]: No need

File: app/api/schema/events.py [hunk 3]
```diff
@@ -15,6 +15,35 @@
...
+        """
+        type_ = 'event-info'
+        self_view = 'v1.event_data_detail'
+        self_view_kwargs = {'id': '<id>'}
+        inflect = dasherize
+
+    id = fields.Str(dump_only=True)
+    has_speakers = fields.Str(allow_none=True, dump_only=True)
+    has_sessions = fields.Str(allow_none=True, dump_only=True)
+    has_tickets = fields.Str(allow_none=True, dump_only=True)
```
Comment (iamareebjamal) [inline app/api/schema/events.py]: Boolean. Why is it string?

**Comment by mrsaicharan1** [comment]
@iamareebjamal Added the appropriate relationships

File: app/models/event.py [hunk 4]
```diff
@@ -35,6 +35,16 @@ def get_new_event_identifier(length=8):
         return get_new_event_identifier(length)
 
 
+class EventData(db.Model):
```
Comment (iamareebjamal) [inline app/models/event.py]: As I told earlier. No DB model is needed

**Comment by iamareebjamal** [comment]
Use this - https://github.com/fossasia/open-event-server/blob/0161e27dcf82f833cf2eec9115de2f5462c8f55d/docs/api/blueprint/event/event_statistics.apib

**Pull request closed without merging**